sdk/core/azure-core/azure/core/tracing/ext/opencensus_wrapper.py

```diff
@@ -0,0 +1,162 @@
+except:
+    AzureExporter = False
+
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, Union
+
+
+class OpencensusSpanWrapper(AbstractSpan):
+    """Wraps a given Opencensus Span so that it implements azure.core.tracing.AbstractSpan"""
+
+    def __init__(self, span=None, name="parent_span"):
+        # type: (Span, str) -> None
+        """
+        If a span is not passed in, creates a new tracer. If the instrumentation key for Azure Exporter is given, will
+        configure the azure exporter else will just create a new tracer.
+
+        :param span: The opencensus span to wrap
+        :param name: The name of the opencensus span to create if a new span is needed
+        """
+        super(OpencensusSpanWrapper, self).__init__(span, name)
+        tracer = self.get_current_tracer()
+        self.was_created_by_azure_sdk = False
+        if span is None and (tracer is None or isinstance(tracer, NoopTracer)):
+            azure_exporter_instrumentation_key = os.environ.get(
+                "APPINSIGHTS_INSTRUMENTATIONKEY", None
+            )
+            prob = float(os.environ.get("AZURE_TRACING_SAMPLER", None) or 0)
+            if azure_exporter_instrumentation_key is not None and AzureExporter:
+                tracer = tracer_module.Tracer(
+                    exporter=AzureExporter(
+                        instrumentation_key=azure_exporter_instrumentation_key
+                    ),
+                    sampler=ProbabilitySampler(prob),
+                )
+            else:
+                tracer = tracer or tracer_module.Tracer(
+                    sampler=ProbabilitySampler(prob)
+                )
+            self.was_created_by_azure_sdk = True
+        span = span or tracer.span(name=name)
+
+        self.tracer = tracer
+        self._span_instance = span
+
+    @property
+    def span_instance(self):
+        # type: () -> Span
+        """
+        :return: The openencensus span that is being wrapped.
+        """
+        return self._span_instance
+
+    def span(self, name="child_span"):
+        # type: (str) -> OpencensusSpanWrapper
+        """
+        Create a child span for the current span and append it to the child spans list in the span instance.
+        :param name: Name of the child span
+        :return: The OpencensusWrapper that is wrapping the child span instance
+        """
+        return self.__class__(self.span_instance.span(name=name))
+
+    def start(self):
+        # type: () -> None
+        """Set the start time for a span."""
+        self.span_instance.start()
+
+    def finish(self):
+        # type: () -> None
+        """Set the end time for a span."""
+        self.span_instance.finish()
+        if self.was_created_by_azure_sdk:
+            self.end_tracer(self.tracer)
+
+    def to_header(self):
+        # type: (Dict[str, str]) -> str
```

Comment:
Type hint and docstring need an update now the parameter's been removed.

Reply:
done